packages/terra-list/src/SelectableList.jsx

```diff
@@ -0,0 +1,158 @@
+  hasChevrons: false,
+  onChange: undefined,
+  selectedIndexes: [],
+};
+
+class SelectableList extends React.Component {
+
+  constructor(props) {
+    super(props);
+    this.handleOnChange = this.handleOnChange.bind(this);
+  }
+
+  handleOnChange(event, index) {
+    if (this.props.onChange) {
+      this.props.onChange(event, index);
+    }
+  }
+
+  cloneChildItems(items) {
+    return React.Children.map(items, (child, index) => {
+      const wrappedOnClick = this.wrappedOnClickForItem(child, index);
+      const wrappedOnKeyDown = this.wrappedOnKeyDownForItem(child, index);
+      const newProps = this.newPropsForItem(child, index, wrappedOnClick, wrappedOnKeyDown);
+
+      return React.cloneElement(child, newProps);
+    });
+  }
+
+  wrappedOnClickForItem(item, index) {
+    const initialOnClick = item.props.onClick;
+
+    return (event) => {
+      // The default isSelectable attribute is either undefined or true, unless the consumer specifies the item isSelectable attribute as false.
+      if (item.props.isSelectable !== false) {
+        this.handleOnChange(event, index);
+      }
+
+      if (initialOnClick) {
+        initialOnClick(event);
+      }
+    };
+  }
+
+  wrappedOnKeyDownForItem(item, index) {
+    const initialOnKeyDown = item.props.onKeyDown;
+
+    return (event) => {
+      if (event.nativeEvent.keyCode === KEYCODES.ENTER || event.nativeEvent.keyCode === KEYCODES.SPACE) {
+        // The default isSelectable attribute is either undefined or true, unless the consumer specifies the item isSelectable attribute as false.
+        if (item.props.isSelectable !== false) {
+          this.handleOnChange(event, index);
+        }
+      }
+
+      if (initialOnKeyDown) {
+        initialOnKeyDown(event);
+      }
+    };
+  }
+
+  newPropsForItem(item, index, onClick, onKeyDown) {
+    const isSelected = this.props.selectedIndexes.indexOf(index) >= 0;
+    const newProps = { };
+
+    // Set the isSelected attribute to false for all the items except the items whose index is set to state selectedIndex
+    if (isSelected !== item.isSelected) {
+      newProps.isSelected = isSelected;
+    }
+
+    // Set the default isSelectable attribute to true, unless the consumer specifies the item isSelectable attribute as false.
+    newProps.isSelectable = true;
+    if (item.props.isSelectable === false) {
+      newProps.isSelectable = item.props.isSelectable;
+    }
+
+    if (this.props.disableUnselectedItems && !isSelected) {
+      newProps.isSelectable = false;
+    }
+
+    // If selectable, add tabIndex on items to navigate through keyboard tab key for selectable lists and add
+    // onClick and onKeyDown functions.
+    if (newProps.isSelectable) {
+      newProps.tabIndex = '0';
+      newProps.onClick = onClick;
+      newProps.onKeyDown = onKeyDown;
+    }
+
+    // Uses the props.hasChevron value, unless the consumer specifies the item hasChevron attribute as false.
+    if (this.props.hasChevrons) {
```

Comment:
I know multi select lists are not intended to render chevrons. Should we add some sort of check to handle that?

Reply:
True, though I'm not sure this agnostic component can distinguish between multi vs single select. Be open to opinions on how this should function if we intend to adopt that behavior.

Reply:
Hmm.. yeah that is a very good point. The cleanest way to enforce this might be to add a boolean prop called `isMultiSelectable` which we could add to the hasChevrons check? That sort-of removes the flexibility of this component, but would kept the distinction between the two? 

Regardless, we should probably update the hasChevrons description to indicated it should only be used when implementing SingleSelect lists.